src/viur/core/module.py

```diff
@@ -1,20 +1,201 @@
-from typing import Dict, Any, Union, Tuple, Callable
+import copy
+import inspect
+import typing
+import logging
+from viur.core import errors, current
 from viur.core.config import conf
 
 
+class Method:
+    """
+    Abstraction wrapper for any public available method.
+    """
+
+    @classmethod
+    def ensure(cls, func):
+        """
+        Ensures the provided `func` parameter is either a Method already, or turns it
+        into a Method. This is done to avoid stacking Method objects, which may create
+        unwanted results.
+        """
+        if isinstance(func, Method):
+            return func
+
+        return cls(func)
+
+    def __init__(self, func: typing.Callable):
+        # Attributes
+        self.exposed = None  # None = unexposed, True = exposed, False = internal exposed
```

Comment:
Can we use a `Enum` here? This is very hacky.

Reply:
I've thought of it first, but then I came to the result that a function should either be "exposed" or not in the future. I really want to get rid of the "internal exposed". I think this will die out with all newer developments we do.